disk.go

```diff
@@ -84,6 +85,55 @@ func (d *Disk) NowIsDone() bool {
 	return true
 }
 
+//==========================================================================
+// Customizable Latency Simulation framework for Disk
+//==========================================================================
+
+// This structure defines the possible parameters that influences the latency
+// function for Disk. Its only queue size now, but will grow in future.
+type DiskLatencyParams struct {
+	qSize	float64
+}
+
+// Interface which implements the latency function
+type DiskLatencySimulator interface {
+	GetName() string
+	Latency(params DiskLatencyParams) time.Duration
+}
+
+type LatencyNil string
+
```

Comment:
1) The latencyNil must be a struct type that implements the DiskLatencySimulator interface. Not sure why it is a string. 
[Vineeth]: I did not have much to add in the structure other than the string. So I just leveraged the go feature that you could add methods to any type. If you feel that structure is cleaner, I will do that.

2) All static declarations must probably move into a single inited struct. 
[Vineeth]: Will do
3) You can always return 0 instead of duration, no need to multiply it. 
[Vineeth]: Will do
4) Golang map can always grow, s/maxLatencySimulators/initialLatencySimulators/
[Vineeth]: Will do
5) Latency(params DiskLatencyParams) time.Duration could be variadic: (args ...interface{}) or at least (args ...float64)
[Vineeth]: I think Structure would be cleaner here. Adding new parameters will anyways need a change in the code, so structure would be more readable
6) Makes sense to add parabolic latency into the code, alongside latency-nil
[Vineeth]: Makes sense. Will do